web/src/pages/api/post/getPost.ts

```diff
@@ -0,0 +1,50 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+import { validateRequest } from "../../../utils/jwt";
+
+interface Request extends NextApiRequest {
+  body: {
+    postId: string;
+  };
+}
+
+export default async function get(req: Request, res: NextApiResponse) {
+  // Validate if the user has a valid JWT token
+  if (!(await validateRequest(req))) {
+    return res.status(401).json({ error: "User not logged in." });
+  }
+
+  const { postId } = req.body;
+
+  if (!postId) {
+    return res
+      .status(400)
+      .json({ error: "You are missing the postId parameter" });
+  }
+
+  if (Array.isArray(postId)) {
+    return res.status(400).json({ error: "postId cannot be a string array" });
+  }
+
+  const post = await prisma.post.findUnique({
+    where: {
+      id: postId,
+    },
+    select: {
+      id: true,
+      dateCreated: true,
+      prompt: true,
+      imageURL: true,
+      author: {
+        select: {
+          name: true,
+          id: true,
+        },
+      },
+      likes: true,
+    },
+  });
+
+  res.json({ post });
```

Comment:
`post` could be null. If it is, do we think it is intuitive to send back `{ post: null }`? We could also consider sending an error. Up to you; maybe check what other people are doing